[WIP]: IBMZ updates for HCP 4.17 (#83206)
openshift/openshift-docs: modules/hcp-ibmz-adding-credentials-registry.adoc

:_mod-docs-content-type: CONCEPT
[id="hcp-ibmz-adding-credentials-registry_{context}"]

> = Adding Credentials and Registry Certificate Authority to the Management Cluster

Reviewer: ```suggestion
= Adding credentials and the registry certificate authority to the management cluster
```

> To pull the mirror registry images from the Management Cluster , you must first add Credentials and Registry Certificate Authority of mirror registry to the Management cluster. Follow the procedure below:

Reviewer: ```suggestion
To pull the mirror registry images from the management cluster, you must first add credentials and the registry certificate authority of the mirror registry to the management cluster.
```
Is the term "registry certificate authority of the mirror registry" redundant? Would it be accurate to simply write, "certificate authority of the mirror registry"?

> 

Reviewer: Before the procedure, add the `.Procedure` heading. See the Procedure template in the modular documentation guidelines for an example: https://github.com/redhat-documentation/modular-docs/blob/main/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc?plain=1

. Create a `ConfigMap` with the certificate of the mirror registry by running the following command:

+
[source,terminal]
----
$ oc apply -f registry-config.yaml
----
+
.Example registry-config.yaml file
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: registry-config
  namespace: openshift-config
data:
  <mirror_registry>: |
    -----BEGIN CERTIFICATE-----
    -----END CERTIFICATE-----
----

> . Patch the cluster-wide object `image.config.openshift.io` to include the following entries:

Reviewer: ```suggestion
. Patch the `image.config.openshift.io` cluster-wide object to include the following entries:
```

+
[source,yaml]
----
spec:
  additionalTrustedCA:
    - name: registry-config
----

> . Update the Management Cluster pull secret to add the credentials of the mirror registry.

Reviewer: ```suggestion
. Update the management cluster pull secret to add the credentials of the mirror registry.
```

> ..  Fetch the exisiting pull secret from the cluster in a JSON format using the following command:

Reviewer: ```suggestion
..  Fetch the pull secret from the cluster in a JSON format by running the following command:
```

+
[source,terminal]
----
$ oc get secret/pull-secret -n openshift-config -o json | jq -r '.data.".dockerconfigjson"' | base64 -d > authfile
----

> .. Edit the fetched secret JSON file to include a section with the credentials of the Registry Certificate Authority: 

Reviewer: ```suggestion
.. Edit the fetched secret JSON file to include a section with the credentials of the certificate authority: 
```

+

> [source,terminal]

Reviewer: In the following section, if the user needs to replace any variables, add callouts to explain what they should replace.

Author: Checked with the team, its a secret pull and we do not want to expose it, at the same time, user doesnt hv to enter these values manually, so removing the <>, to indicate that there is no user action required here.

Author: Done

----
  "auths": {
    "<mirror_registry>": { 
      "auth": "<credentials>", 
      "email": "you@example.com"
    }
  },
----

> .. Update the pull secret on the cluster using the following command:

Reviewer: ```suggestion
.. Update the pull secret on the cluster by running the following command:
```
